### PR TITLE
Support multiple translation providers

### LINKED ISSUE #8939: Support multiple translation providers
Localization manager should be able to handle multiple instances of ITranslationProvider. This way we can create more methods for storing translations.

### PR DESCRIPTION
**Pull request by doticca**
Adds support to LocalizationManager for multiple translation providers

Fixes #8939

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/OrchardCMS/OrchardCore?pullRequest=8940) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/OrchardCMS/OrchardCore?pullRequest=8940) before we can accept your contribution.<br/><br/>:x: doticca [sign now](https://cla.dotnetfoundation.org/OrchardCMS/OrchardCore?pullRequest=8940)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/OrchardCMS/OrchardCore?pullRequest=8940) it.</sub>

**Comment by Skrypt** [comment]
Related with https://github.com/OrchardCMS/OrchardCore/pull/8098

File: src/OrchardCore/OrchardCore.Localization.Core/LocalizationManager.cs [hunk 1]
```diff
@@ -52,7 +52,10 @@ public CultureDictionary GetDictionary(CultureInfo culture)
                 }
 
                 var dictionary = new CultureDictionary(culture.Name, rule ?? DefaultPluralRule);
-                _translationProvider.LoadTranslations(culture.Name, dictionary);
+                foreach (var translationProvider in this._translationProviders)
```
Comment (hishamco) [inline src/OrchardCore/OrchardCore.Localization.Core/LocalizationManager.cs:55]: Just remove `this`

**Review by hishamco — CHANGES_REQUESTED** [review]
LGTM except the little comment

**Comment by Skrypt** [comment]
Ok, I just analyzed this and it makes sense. Now we have the PoFilesTranslationsProvider and DataTranslationsProvider as ITranslationProviders. They are 2 different providers so that they return different localizations results. Basically, you should never use `T["Something"]` to provide a Content Type/Dynamic data name translation but `D["Something"]`. If we would have only one ITranslationProvider it would probably merge translations together in the same IDictionary.

**Pull request merged**